### PR TITLE
add: previous master-workers schedule

### PR DESCRIPTION
**Pull request by ActivePeter**

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -28,13 +28,23 @@ pub struct NodeConfig {
     pub addr: SocketAddr,
     pub spec: HashSet<String>,
 }
+
```
Comment (NaturalSelect) [inline src/config.rs:31]: Why choose this design?

File: src/schedule/container_manager.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                let _ = self.cache.remove(&k);
+            }
+        }
+        let _ = self.cache.insert(key.to_string(), value);
+        self.lru.push_front(key.to_string());
+    }
+}
+
+pub struct ContainerManager {
+    cache: Mutex<LRUCache<Vm>>,
```
Comment (NaturalSelect) [inline src/schedule/container_manager.rs:50]: Use an indirect cache layer instead of `LRUCache`?

Try S3-FIFO?

**Review by NaturalSelect — APPROVED** [review]
LGTM.

File: src/schedule/container_manager.rs (hunk 2 above)
Comment (ActivePeter) [reply src/schedule/container_manager.rs:50]: We can add more cache rules and do comprehensive tests on them in future works.

File: src/config.rs (hunk 1 above)
Comment (ActivePeter) [reply src/config.rs:31]: Just simple for now.

**Pull request merged**